.github/workflows/package.yaml

```diff
@@ -54,103 +55,72 @@ jobs:
+      if: matrix.platform == 'mac'
       with:
-        name: Rancher Desktop-mac.aarch64.zip
+        name: Rancher Desktop-mac.${{ matrix.arch }}.zip
         path: dist/Rancher Desktop*.zip
         if-no-files-found: error
-    - uses: actions/upload-artifact@v3
-      if: matrix.os == 'windows'
+    - name: Upload Windows installer
+      uses: actions/upload-artifact@v3
+      if: matrix.platform == 'win'
       with:
         name: Rancher Desktop Setup.exe
         path: dist/Rancher Desktop*.exe
         if-no-files-found: error
-    - uses: actions/upload-artifact@v3
-      if: matrix.os == 'windows'
+    - name: Upload Windows zip
+      uses: actions/upload-artifact@v3
+      if: matrix.platform == 'win'
       with:
         name: Rancher Desktop-win.zip
         path: dist/Rancher Desktop-*-win.zip
         if-no-files-found: error
-    - name: set zip_name env var
-      run: |
-        # in pull requests GITHUB_REF_NAME is in the form "<pr_number>/merge";
-        # remove slashes since they aren't valid in filenames
-        no_slash_ref_name=$(echo $GITHUB_REF_NAME | sed -E 's/\//-/g')
-        zip_name="rancher-desktop-linux-${no_slash_ref_name}.zip"
-        echo "zip_name=$zip_name" >> $GITHUB_ENV
-    - name: Ensure linux zip file has a known name
-      uses: canastro/copy-file-action@0.0.2
-      if: matrix.os == 'linux'
-      with:
-        source: "dist/rancher-desktop*.zip"
-        target: ${{ format('dist/{0}', env.zip_name) }}
-    - name: Upload zip as workflow artifact
+    - name: Upload Linux zip
       uses: actions/upload-artifact@v3
-      if: matrix.os == 'linux'
+      if: matrix.platform == 'linux'
       with:
         name: Rancher Desktop-linux.zip
-        path: ${{ format('dist/{0}', env.zip_name) }}
+        path: dist/rancher-desktop-*-linux.zip
         if-no-files-found: error
     - id: has_s3
       name: Check if S3 secrets are available
       continue-on-error: true
-      if: matrix.os == 'linux' && github.ref_type == 'branch' && ( startsWith(github.ref_name, 'main') || startsWith(github.ref_name, 'release-') )
+      if: github.ref_type == 'branch' && ( startsWith(github.ref_name, 'main') || startsWith(github.ref_name, 'release-') )
       run: '[[ -n "${key}" ]]'
       env:
         key: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    - name: set zip_name env var
+      id: zip_name
+      if: matrix.platform == 'linux' && steps.has_s3.outcome == 'success'
+      run: |
+        # in pull requests GITHUB_REF_NAME is in the form "<pr_number>/merge";
+        # remove slashes since they aren't valid in filenames
+        no_slash_ref_name="${GITHUB_REF_NAME//\//-/}"
+        zip_name="rancher-desktop-linux-${no_slash_ref_name}.zip"
+        echo "::set-output name=zip_name::${zip_name}"
     - name: Copy zip file to S3
-      uses: prewk/s3-cp-action@v2
-      if: steps.has_s3.outcome == 'success'
+      uses: prewk/s3-cp-action@74701625561055a306f92fa5c18e948f9d14a54a
```

Comment:
What is the reason to pin the action to this specific commit? Because the tag could be redefined, and this step includes sensitive credentials?